### PR TITLE
Add JSON Documentation & Document Generator 

### PR DESCRIPTION
**Pull request by noursaidi**
Adds document generator to convert JSON schema into HTML files. Also included the HTML version of the current schema.

Preview - https://noursaidi.github.io/udmi/schema/index.html 

Suggest that GitHub Pages is enabled on the docs directory to allow it to be viewed nicely from the web.

### CODE REVIEW

**Comment by noursaidi** [comment]
I've now added the more basic descriptions to this PR and many of the descriptions which already exist in the current documentation

File: validator_config.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (grafnu) [inline validator_config.json:1]: this file should not be comitted!  Likely need to add it to .gitignore or similar.

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -71,3 +71,11 @@ jobs:
...
           echo Pubber output logs:
           cat pubber.out || true
+  
+  docs:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v2
+      - name: schema docs
+        run: bin/gendocs check
```
Comment (grafnu) [inline .github/workflows/testing.yml:81]: newline at end of file!

**Review by grafnu — COMMENTED** [review]
Just some pre-review comments.

Also, what do you think about relocating the files to gencode/docs rather than docs/schema?  The reason is that it's nice to keep everything that's automatically generated in one place so it doesn't get mixed in with manual edits. Right now, everything else in gencode/ is automatically generated, so seems reasonable.

**Comment by noursaidi** [comment]
Resolved pre-review comments and moved directory into gencode/docs. Most of the more straightforward definitions which already exist in the current documentation have been copied over, rest of definitions to come later.

New web preview link - https://noursaidi.github.io/udmi/gencode/docs/

File: bin/gendocs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#RELATIVE_LINK="..\/"
+#find $OUTPUT_DIR -type f -exec sed -i '' "s/$ABSOLUTE_LINK/$RELATIVE_LINK/" {} \;
+#ABSOLUTE_LINK="https:\/\/github.com\/faucetsdn\/udmi\/blob\/master\/docs\/"
+
+# Create index page with links to different schemas
+cp etc/schema_index_template.html $OUTPUT_DIR/index.html
+
+for schema in $schemas; do
+    sed -i '' "s/<!--SCHEMALINK-->/\
+              <div class=\"card\"><a href=\"$schema.html\">$schema<\/a><\/div><!--SCHEMALINK-->/"\
```
Comment (grafnu) [inline bin/gendocs:51]: nit: typically indent would be hanging-4, not vertically aligned.

**Review by grafnu — APPROVED** [review]

**Comment by noursaidi** [comment]
Indent fixed - ready to be merged now

**Pull request merged**